src/ProgressOnderwijsUtils.Analyzers/UseAsEnrolledSqlInExpressionAnalyzer.cs

```diff
@@ -0,0 +1,68 @@
+using System.Linq;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace ProgressOnderwijsUtils.Analyzers;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+public sealed class UseAsEnrolledSqlInExpressionAnalyzer : DiagnosticAnalyzer
+{
+    public static readonly DiagnosticDescriptor Rule = new(
+#pragma warning disable RS2008 // Enable analyzer release tracking - why would we care?
+        "POU1010",
+#pragma warning restore RS2008 // Enable analyzer release tracking
+        "Use IEnumerable<>.AsEnrolledSqlInExpression()",
+        "Use the more efficient IEnumerable<>.AsEnrolledSqlInExpression() for in expressions on small sets of constant size.",
+        "Functional",
+        DiagnosticSeverity.Error,
+        true
+    );
+
+    public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
+        => ImmutableArray.Create(Rule);
+
+    public override void Initialize(AnalysisContext context)
+    {
+        context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze);
+        context.EnableConcurrentExecution();
+        context.RegisterSyntaxNodeAction(
+            Analyze,
+            SyntaxKind.InvocationExpression
+        );
+    }
+
+    static void Analyze(SyntaxNodeAnalysisContext context)
+    {
+        if (context.Node is InvocationExpressionSyntax expr
+            && context.SemanticModel.GetSymbolInfo(expr).Symbol is IMethodSymbol { Name: "SQL" } method
```

Comment:
semantic models zijn vrij duur, dus om VS niet te vertragen wil je zoveel mogelijk filtering alvast gedaan hebben voordat je deze opvraagt.

Tevens is het verstandig een build-benchmark vooraf en achteraf te doen, just to be sure dat dit geen hoge overhead heeft.